tests/login.go

```diff
@@ -113,49 +110,28 @@ func LoggedInFedoraExpecter(vmi *v1.VirtualMachineInstance) (expect.Expecter, er
 	}
 	b := append([]expect.Batcher{
 		&expect.BSnd{S: "\n"},
-		&expect.BSnd{S: "\n"},
-		&expect.BCas{C: []expect.Caser{
-			&expect.Case{
-				// Using only "login: " would match things like "Last failed login: Tue Jun  9 22:25:30 UTC 2020 on ttyS0"
-				// and in case the VM's did not get hostname form DHCP server try the default hostname
-				R:  regexp.MustCompile(fmt.Sprintf(`(localhost|%s) login: `, vmi.Name)),
-				S:  "fedora\n",
-				T:  expect.Next(),
-				Rt: 10,
-			},
-			&expect.Case{
-				R:  regexp.MustCompile(`Password:`),
-				S:  "fedora\n",
-				T:  expect.Next(),
-				Rt: 10,
-			},
-			&expect.Case{
-				R:  regexp.MustCompile(`Login incorrect`),
-				T:  expect.LogContinue("Failed to log in", expect.NewStatus(codes.PermissionDenied, "login failed")),
-				Rt: 10,
-			},
-			&expect.Case{
-				R: regexp.MustCompile(`\$ `),
-				T: expect.OK(),
-			},
-		}},
+		&expect.BExp{R: fmt.Sprintf(`(localhost|%s) login: `, vmi.Name)},
```

Comment:
There's a good reason for the Caser above. Sometimes we try to login too early, and credentials are not yet setup, which leads to test flakyness.

Reply:
I understand, but I think it didn't work as expected, it matched the last case - the prompt - even though it didn't actually reached the prompt. it found a `$` (or a `#`) in the login log(in the ssh key image for example) , and "matched" though it didn't ever got to the login prompt. so you think there is another way to address this issue without  giving up on the cases?

Another thing to consider is that maybe the flakyness you are talking  was caused by the issue I mentioned above? I ran those tests many times and I don't think I saw the issue you are talking about

Reply:
Good catch for the `$ ` matching outside of login, we do need to fix that.
I'm not sure which issue you're referring to, but the reason we need the Caser is that we need to wait until cloud-init is done, which may happen after the login prompt.
Can we keep the Caser and expect the full prompt instead? (`[fedora@vmi-fedora ~]$ `)

Reply:
> Can we keep the Caser and expect the full prompt instead? (`[fedora@vmi-fedora ~]$ `)

I think that will  work, but will will loose the generalization we hoped to achieve with the prompt expression.

maybe we can check the  /var/lib/cloud/instance/boot-finished file for existence. something like: 

```
		&expect.BSnd{S: "ls /var/lib/cloud/instance/boot-finished\n"},
		&expect.BExp{R: PromptExpression},
		&expect.BSnd{S: "echo $?\n"},
		&expect.BExp{R: RetValue("0")},
```
This file appears when cloud-init finishes boot. I'll have to verify it, but this should work. what do you think?

Reply:
Well you have to be logged-in for that no? :)

Reply:
haha ooops.
Right.
So I'll use the explicit fedora prompt.

Reply:
Assumed yes and returned the cases with the full prompt expected in the last case.